### ISSUE TITLE
runqlen.bt is not working in current development version of bpftrace

### ISSUE THREAD

**Issue by caringi** [iovisor/bpftrace#752]
I didn't investigate much the problem, but it seems related to clang struct parsing...

```
# ./bpftrace  ../../tools/runqlen.bt 
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:209:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:255:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:308:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/jump_label.h:23:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/jump_label.h:39:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:85:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:125:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:139:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:154:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:76:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:118:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:132:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:147:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:550:17: warning: taking address of packed member 'sp0' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/cpufeature.h:164:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/preempt.h:94:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/qspinlock.h:23:8: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qspinlock.h:74:55: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qspinlock.h:87:52: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qrwlock.h:74:56: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qrwlock.h:101:53: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
# echo $?
1
```

Using bpftrace v0.9 I get similar warning messages, but it does work:

```
bpftrace ../../tools/runqlen.bt 
/bpftrace/include/stdarg.h:52:1: warning: null character ignored [-Wnull-character]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:209:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:255:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/bitops.h:308:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/jump_label.h:23:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/jump_label.h:39:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:85:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:125:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:139:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic.h:154:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:76:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:118:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:132:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/atomic64_64.h:147:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:550:17: warning: taking address of packed member 'sp0' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/processor.h:567:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an unaligned pointer value [-Waddress-of-packed-member]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/cpufeature.h:164:2: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/preempt.h:94:9: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/arch/x86/include/asm/qspinlock.h:23:8: error: 'asm goto' constructs are not supported yet
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qspinlock.h:74:55: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qspinlock.h:87:52: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qrwlock.h:74:56: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/qrwlock.h:101:53: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
/lib/modules/5.0.17-300.fc30.x86_64+debug/source/include/asm-generic/atomic-instrumented.h:69:66: note: passing argument to parameter 'old' here
Attaching 2 probes...
Sampling run queue length at 99 Hertz... Hit Ctrl-C to end.
^C

@runqlen: 
[0, 1)              1030 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@|
[1, 2)                 3 |                                                    |
[2, 3)                 2 |                                                    |
[3, 4)                 0 |                                                    |
[4, 5)                 1 |                                                    |


```

**Comment by caringi** [comment]
Complementing:

It started to not work after ecb7a1bf422fe251fc16ac65405d4a6c2cbb927c
But with an error message:

`Can't get size of 'struct mm_struct::cpu_bitmap', please provide proper definiton.`

Then with 22397560daaf8cad259540c303d3e3d5a2f27c69 it started to work again...

And finally to not work with any specific error message after 364849d702792090e7eab9e0474cfb5d1097603a

**Comment by mmarchini** [comment]
Will be fixed with #732.

**Comment by mmarchini** [comment]
Closing since it should be fixed now. Feel free to reopen if that's still an issue.
